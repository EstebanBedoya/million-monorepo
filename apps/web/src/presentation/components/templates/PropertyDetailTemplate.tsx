'use client';

import { ArrowLeft } from 'lucide-react';
import { MockPropertyType } from '../../../domain/schemas/property.schema';
import { Button } from '../atoms/Button';
import { Icon } from '../atoms/Icon';
import { Image } from '../atoms/Image';
import { Price } from '../atoms/Price';
import { Badge } from '../atoms/Badge';
import { Breadcrumbs, BreadcrumbItem } from '../molecules/Breadcrumbs';
import { LocationInfo } from '../molecules/LocationInfo';
import { PropertyDetails } from '../molecules/PropertyDetails';

export interface PropertyDetailTemplateProps {
  property: MockPropertyType;
  onBack: () => void;
}

export const PropertyDetailTemplate = ({ property, onBack }: PropertyDetailTemplateProps) => {
  const breadcrumbItems: BreadcrumbItem[] = [
    { label: 'Home', href: '/' },
    { label: 'Properties', href: '/' },
    { label: property.name }
  ];

  return (
    <div className="min-h-screen bg-background">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8">
        <div className="mb-6">
          <Button 
            variant="ghost" 
            onClick={onBack}
            className="mb-4 -ml-2"
            aria-label="Go back to properties list"
          >
            <Icon icon={ArrowLeft} size="sm" className="mr-2" />
            Back
          </Button>
          
          <Breadcrumbs items={breadcrumbItems} />
        </div>

        <div className="grid grid-cols-1 lg:grid-cols-3 gap-8">
          <div className="lg:col-span-2 space-y-6">
            <div className="relative h-[400px] md:h-[500px] rounded-lg overflow-hidden bg-secondary/10">
              <Image
                src={property.image || '/placeholder-property.jpg'}
                alt={property.name}
                fill
                priority
                sizes="(max-width: 768px) 100vw, 66vw"
                className="object-cover"
                fallbackSrc="/placeholder-property.jpg"
              />
              
              {property.propertyType && (
                <div className="absolute top-4 left-4">
                  <Badge variant="secondary" className="bg-gray-700/50 text-gray-100 backdrop-blur-sm shadow-sm">
                    {property.propertyType}
                  </Badge>
                </div>
              )}
            </div>

            <div>
              <h1 className="text-3xl md:text-4xl font-bold text-foreground mb-4">
                {property.name}
              </h1>

              <div className="mb-6">
                <Price amount={property.price} size="2xl" className="mb-3" />
                <LocationInfo 
                  address={property.address}
                  city={property.city}
                />
              </div>

              <PropertyDetails
                bedrooms={property.bedrooms}
                bathrooms={property.bathrooms}
                area={property.area}
                areaUnit={property.areaUnit}
                className="p-4 bg-card border border-border rounded-lg"
              />
            </div>

            <div className="card-elevated p-6">
              <h2 className="text-2xl font-semibold text-foreground mb-4">
                Description
              </h2>
              <p className="text-secondary leading-relaxed">
                This beautiful {property.propertyType?.toLowerCase() || 'property'} is located in {property.city}, 
                offering {property.bedrooms ? `${property.bedrooms} bedroom${property.bedrooms > 1 ? 's' : ''}` : 'spacious living areas'} 
                {property.bathrooms ? ` and ${property.bathrooms} bathroom${property.bathrooms > 1 ? 's' : ''}` : ''}
                {property.area ? ` with ${property.area} ${property.areaUnit || 'm²'} of living space` : ''}.
                Perfect for those seeking a comfortable and elegant living experience in a prime location.
              </p>
            </div>

            <div className="card-elevated p-6">
              <h2 className="text-2xl font-semibold text-foreground mb-4">
                Property Details
              </h2>
              <dl className="grid grid-cols-1 md:grid-cols-2 gap-4">
                <div className="border-b border-border pb-3">
                  <dt className="text-sm text-secondary mb-1">Property ID</dt>
                  <dd className="font-medium text-foreground">{property.id}</dd>
                </div>
                
                {property.propertyType && (
                  <div className="border-b border-border pb-3">
                    <dt className="text-sm text-secondary mb-1">Property Type</dt>
                    <dd className="font-medium text-foreground">{property.propertyType}</dd>
                  </div>
                )}
                
                <div className="border-b border-border pb-3">
                  <dt className="text-sm text-secondary mb-1">Location</dt>
                  <dd className="font-medium text-foreground">{property.city}</dd>
                </div>
                
                <div className="border-b border-border pb-3">
                  <dt className="text-sm text-secondary mb-1">Address</dt>
                  <dd className="font-medium text-foreground">{property.address}</dd>
                </div>

                {property.bedrooms !== undefined && (
                  <div className="border-b border-border pb-3">
                    <dt className="text-sm text-secondary mb-1">Bedrooms</dt>
                    <dd className="font-medium text-foreground">{property.bedrooms}</dd>
                  </div>
                )}

                {property.bathrooms !== undefined && (
                  <div className="border-b border-border pb-3">
                    <dt className="text-sm text-secondary mb-1">Bathrooms</dt>
                    <dd className="font-medium text-foreground">{property.bathrooms}</dd>
                  </div>
                )}

                {property.area && (
                  <div className="border-b border-border pb-3">
                    <dt className="text-sm text-secondary mb-1">Total Area</dt>
                    <dd className="font-medium text-foreground">{property.area} {property.areaUnit || 'm²'}</dd>
                  </div>
                )}
              </dl>
            </div>
          </div>

          <div className="lg:col-span-1">
            <div className="card-elevated p-6 sticky top-8">
              <div className="mb-6">
                <Price amount={property.price} size="2xl" />
              </div>

              <div className="space-y-4 mb-6">
                <div className="flex items-center justify-between py-3 border-b border-border">
                  <span className="text-sm text-secondary">Property Type</span>
                  <span className="font-medium text-foreground">{property.propertyType || 'N/A'}</span>
                </div>
                
                <div className="flex items-center justify-between py-3 border-b border-border">
                  <span className="text-sm text-secondary">Location</span>
                  <span className="font-medium text-foreground">{property.city}</span>
                </div>

                {property.area && (
                  <div className="flex items-center justify-between py-3 border-b border-border">
                    <span className="text-sm text-secondary">Area</span>
                    <span className="font-medium text-foreground">{property.area} {property.areaUnit || 'm²'}</span>
                  </div>
                )}
              </div>

              <Button 
                className="w-full mb-3"
                aria-label="Contact about this property"
              >
                Contact Agent
              </Button>

              <Button 
                variant="outline" 
                className="w-full"
                aria-label="Schedule a viewing"
              >
                Schedule Viewing
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

