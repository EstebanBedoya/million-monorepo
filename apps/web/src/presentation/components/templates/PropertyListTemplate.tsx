'use client';

import { MockPropertyType } from '../../../domain/schemas/property.schema';
import { PropertyCard } from '../organisms/PropertyCard';
import { AsyncBoundary } from '../molecules/AsyncBoundary';

interface PropertyListTemplateProps {
  properties: MockPropertyType[];
  onPropertyClick?: (property: MockPropertyType) => void;
  loading?: boolean;
  error?: Error | string | null;
  onRetry?: () => void;
  onClearFilters?: () => void;
}

export function PropertyListTemplate({ 
  properties, 
  onPropertyClick, 
  loading,
  error,
  onRetry,
  onClearFilters 
}: PropertyListTemplateProps) {
  return (
    <AsyncBoundary
      loading={loading}
      error={error}
      isEmpty={!loading && !error && properties.length === 0}
      onRetry={onRetry}
      onClearFilters={onClearFilters}
      emptyTitle="No Properties Found"
      emptyMessage="We couldn't find any properties matching your criteria. Try adjusting your filters."
    >
      <div 
        className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-6"
        role="list"
        aria-label="Properties list"
      >
        {properties.map((property) => (
          <div key={property.id} role="listitem">
            <PropertyCard
              property={property}
              onViewDetails={onPropertyClick}
            />
          </div>
        ))}
      </div>
    </AsyncBoundary>
  );
}
