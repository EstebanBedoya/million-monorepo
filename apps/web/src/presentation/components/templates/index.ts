export { PropertyListTemplate } from './PropertyListTemplate';
export { FiltersTemplate } from './FiltersTemplate';
export { PropertyDetailTemplate } from './PropertyDetailTemplate';
