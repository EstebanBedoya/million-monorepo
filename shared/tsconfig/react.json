{
  "extends": "./base.json",
  "compilerOptions": {
    "lib": ["ES2022", "DOM", "DOM.Iterable"],
    "jsx": "react-jsx",
    "allowJs": true,
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ]
  }
}
