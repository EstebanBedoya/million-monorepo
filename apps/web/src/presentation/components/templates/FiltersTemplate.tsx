'use client';

import { ReactNode } from 'react';

interface FiltersTemplateProps {
  children: ReactNode;
  className?: string;
}

export function FiltersTemplate({ children, className }: FiltersTemplateProps) {
  return (
    <div className={`bg-card border border-border rounded-lg p-6 shadow-sm mb-8 ${className}`}>
      {children}
    </div>
  );
}
