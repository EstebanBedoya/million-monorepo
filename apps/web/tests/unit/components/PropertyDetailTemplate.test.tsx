import { render, screen } from '@testing-library/react';
import { PropertyDetailTemplate } from '../../../src/presentation/components/templates/PropertyDetailTemplate';
import { MockPropertyType } from '../../../src/domain/schemas/property.schema';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
    return <img {...props} />;
  },
}));

describe('PropertyDetailTemplate', () => {
  const mockProperty: MockPropertyType = {
    id: 'prop-789',
    idOwner: 'owner-002',
    name: 'Luxury Penthouse',
    address: '789 Park Avenue',
    city: 'New York',
    price: 8500000,
    image: 'https://example.com/penthouse.jpg',
    bedrooms: 4,
    bathrooms: 3,
    area: 300,
    areaUnit: 'mÂ²',
    propertyType: 'Apartment',
  };

  const mockOnBack = jest.fn();

  beforeEach(() => {
    mockOnBack.mockClear();
  });

  it('should render property details correctly', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    expect(screen.getByRole('heading', { name: 'Luxury Penthouse' })).toBeInTheDocument();
    expect(screen.getAllByText('$8,500,000').length).toBeGreaterThan(0);
    expect(screen.getAllByText('New York').length).toBeGreaterThan(0);
    expect(screen.getAllByText('789 Park Avenue').length).toBeGreaterThan(0);
  });

  it('should render breadcrumbs', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    const breadcrumb = screen.getByRole('navigation', { name: /Breadcrumb/i });
    expect(breadcrumb).toBeInTheDocument();
    
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Properties')).toBeInTheDocument();
  });

  it('should render back button', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    const backButton = screen.getByRole('button', { name: /Go back to properties list/i });
    expect(backButton).toBeInTheDocument();
  });

  it('should call onBack when back button is clicked', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    const backButton = screen.getByRole('button', { name: /Go back to properties list/i });
    backButton.click();

    expect(mockOnBack).toHaveBeenCalledTimes(1);
  });

  it('should render property type badge', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    const badges = screen.getAllByText('Apartment');
    expect(badges.length).toBeGreaterThan(0);
  });

  it('should render contact and schedule buttons', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    expect(screen.getByRole('button', { name: /Contact about this property/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /Schedule a viewing/i })).toBeInTheDocument();
  });

  it('should render property details section', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    expect(screen.getByText('Property Details')).toBeInTheDocument();
    expect(screen.getByText('Property ID')).toBeInTheDocument();
    expect(screen.getByText('prop-789')).toBeInTheDocument();
  });

  it('should render description section', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    expect(screen.getByText('Description')).toBeInTheDocument();
  });

  it('should match snapshot with full property data', () => {
    const { container } = render(
      <PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with minimal property data', () => {
    const minimalProperty: MockPropertyType = {
      id: 'prop-minimal',
      idOwner: 'owner-003',
      name: 'Simple House',
      address: '123 Simple St',
      city: 'Simple City',
      price: 250000,
      image: 'https://example.com/simple.jpg',
    };

    const { container } = render(
      <PropertyDetailTemplate property={minimalProperty} onBack={mockOnBack} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render without property type', () => {
    const propertyWithoutType: MockPropertyType = {
      ...mockProperty,
      propertyType: undefined,
    };

    render(<PropertyDetailTemplate property={propertyWithoutType} onBack={mockOnBack} />);

    expect(screen.getByRole('heading', { name: 'Luxury Penthouse' })).toBeInTheDocument();
  });

  it('should render property details in sidebar', () => {
    render(<PropertyDetailTemplate property={mockProperty} onBack={mockOnBack} />);

    const priceElements = screen.getAllByText('$8,500,000');
    expect(priceElements.length).toBeGreaterThan(0);
  });
});

