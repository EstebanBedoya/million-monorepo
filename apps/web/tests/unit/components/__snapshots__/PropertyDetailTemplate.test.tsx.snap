// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyDetailTemplate should match snapshot with full property data 1`] = `
<div
  class="min-h-screen bg-background"
>
  <div
    class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8"
  >
    <div
      class="mb-6"
    >
      <button
        aria-label="Go back to properties list"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed text-foreground hover:bg-secondary/5 focus:ring-accent h-10 px-4 text-sm mb-4 -ml-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left w-4 h-4 text-current mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        Back
      </button>
      <nav
        aria-label="Breadcrumb"
      >
        <ol
          class="flex items-center space-x-2 flex-wrap"
        >
          <li
            class="flex items-center"
          >
            <a
              class="text-sm text-secondary hover:text-accent transition-colors"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right w-4 h-4 text-secondary mx-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <a
              class="text-sm text-secondary hover:text-accent transition-colors"
              href="/"
            >
              Properties
            </a>
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right w-4 h-4 text-secondary mx-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <span
              aria-current="page"
              class="text-sm font-medium text-foreground"
            >
              Luxury Penthouse
            </span>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-3 gap-8"
    >
      <div
        class="lg:col-span-2 space-y-6"
      >
        <div
          class="relative h-[400px] md:h-[500px] rounded-lg overflow-hidden bg-secondary/10"
        >
          <img
            alt="Luxury Penthouse"
            class="object-cover"
            sizes="(max-width: 768px) 100vw, 66vw"
            src="https://example.com/penthouse.jpg"
          />
          <div
            class="absolute top-4 left-4"
          >
            <div
              class="inline-flex items-center rounded-md font-medium px-3 py-1.5 text-sm bg-gray-700/50 text-gray-100 backdrop-blur-sm shadow-sm"
            >
              Apartment
            </div>
          </div>
        </div>
        <div>
          <h1
            class="text-3xl md:text-4xl font-bold text-foreground mb-4"
          >
            Luxury Penthouse
          </h1>
          <div
            class="mb-6"
          >
            <span
              class="font-semibold text-accent font-cairo mb-3"
            >
              $8,500,000
            </span>
            <div
              class="flex items-start gap-2 undefined"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-map-pin w-5 h-5 text-secondary flex-shrink-0 mt-0.5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
              <div
                class="flex-1 min-w-0"
              >
                <p
                  class="text-sm text-secondary line-clamp-2"
                >
                  789 Park Avenue
                </p>
                <p
                  class="text-sm font-medium text-foreground mt-0.5"
                >
                  New York
                </p>
              </div>
            </div>
          </div>
          <div
            class="flex items-center gap-4 pt-4 border-t border-border p-4 bg-card border border-border rounded-lg"
          >
            <div
              class="flex items-center gap-1.5"
              title="4 Bedrooms"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-house w-4 h-4 text-secondary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 21v-8a1 1 0 0 0-1-1h-4a1 1 0 0 0-1 1v8"
                />
                <path
                  d="M3 10a2 2 0 0 1 .709-1.528l7-6a2 2 0 0 1 2.582 0l7 6A2 2 0 0 1 21 10v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                />
              </svg>
              <span
                class="text-sm text-foreground font-medium"
              >
                4
                 
                <span
                  class="text-secondary font-normal"
                >
                  bd
                </span>
              </span>
            </div>
            <div
              class="flex items-center gap-1.5"
              title="3 Bathrooms"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-building w-4 h-4 text-secondary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 10h.01"
                />
                <path
                  d="M12 14h.01"
                />
                <path
                  d="M12 6h.01"
                />
                <path
                  d="M16 10h.01"
                />
                <path
                  d="M16 14h.01"
                />
                <path
                  d="M16 6h.01"
                />
                <path
                  d="M8 10h.01"
                />
                <path
                  d="M8 14h.01"
                />
                <path
                  d="M8 6h.01"
                />
                <path
                  d="M9 22v-3a1 1 0 0 1 1-1h4a1 1 0 0 1 1 1v3"
                />
                <rect
                  height="20"
                  rx="2"
                  width="16"
                  x="4"
                  y="2"
                />
              </svg>
              <span
                class="text-sm text-foreground font-medium"
              >
                3
                 
                <span
                  class="text-secondary font-normal"
                >
                  ba
                </span>
              </span>
            </div>
            <div
              class="flex items-center gap-1.5"
              title="300 m²"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-maximize w-4 h-4 text-secondary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 3H5a2 2 0 0 0-2 2v3"
                />
                <path
                  d="M21 8V5a2 2 0 0 0-2-2h-3"
                />
                <path
                  d="M3 16v3a2 2 0 0 0 2 2h3"
                />
                <path
                  d="M16 21h3a2 2 0 0 0 2-2v-3"
                />
              </svg>
              <span
                class="text-sm text-foreground font-medium"
              >
                300
                 
                <span
                  class="text-secondary font-normal"
                >
                  m²
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="card-elevated p-6"
        >
          <h2
            class="text-2xl font-semibold text-foreground mb-4"
          >
            Description
          </h2>
          <p
            class="text-secondary leading-relaxed"
          >
            This beautiful 
            apartment
             is located in 
            New York
            , offering 
            4 bedrooms
             and 3 bathrooms
             with 300 m² of living space
            . Perfect for those seeking a comfortable and elegant living experience in a prime location.
          </p>
        </div>
        <div
          class="card-elevated p-6"
        >
          <h2
            class="text-2xl font-semibold text-foreground mb-4"
          >
            Property Details
          </h2>
          <dl
            class="grid grid-cols-1 md:grid-cols-2 gap-4"
          >
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Property ID
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                prop-789
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Property Type
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                Apartment
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Location
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                New York
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Address
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                789 Park Avenue
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Bedrooms
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                4
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Bathrooms
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                3
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Total Area
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                300
                 
                m²
              </dd>
            </div>
          </dl>
        </div>
      </div>
      <div
        class="lg:col-span-1"
      >
        <div
          class="card-elevated p-6 sticky top-8"
        >
          <div
            class="mb-6"
          >
            <span
              class="font-semibold text-accent font-cairo"
            >
              $8,500,000
            </span>
          </div>
          <div
            class="space-y-4 mb-6"
          >
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <span
                class="text-sm text-secondary"
              >
                Property Type
              </span>
              <span
                class="font-medium text-foreground"
              >
                Apartment
              </span>
            </div>
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <span
                class="text-sm text-secondary"
              >
                Location
              </span>
              <span
                class="font-medium text-foreground"
              >
                New York
              </span>
            </div>
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <span
                class="text-sm text-secondary"
              >
                Area
              </span>
              <span
                class="font-medium text-foreground"
              >
                300
                 
                m²
              </span>
            </div>
          </div>
          <button
            aria-label="Contact about this property"
            class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed bg-accent text-white hover:bg-accent/90 focus:ring-accent h-10 px-4 text-sm w-full mb-3"
          >
            Contact Agent
          </button>
          <button
            aria-label="Schedule a viewing"
            class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-10 px-4 text-sm w-full"
          >
            Schedule Viewing
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyDetailTemplate should match snapshot with minimal property data 1`] = `
<div
  class="min-h-screen bg-background"
>
  <div
    class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-8"
  >
    <div
      class="mb-6"
    >
      <button
        aria-label="Go back to properties list"
        class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed text-foreground hover:bg-secondary/5 focus:ring-accent h-10 px-4 text-sm mb-4 -ml-2"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left w-4 h-4 text-current mr-2"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        Back
      </button>
      <nav
        aria-label="Breadcrumb"
      >
        <ol
          class="flex items-center space-x-2 flex-wrap"
        >
          <li
            class="flex items-center"
          >
            <a
              class="text-sm text-secondary hover:text-accent transition-colors"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right w-4 h-4 text-secondary mx-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <a
              class="text-sm text-secondary hover:text-accent transition-colors"
              href="/"
            >
              Properties
            </a>
          </li>
          <li
            class="flex items-center"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-right w-4 h-4 text-secondary mx-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
            <span
              aria-current="page"
              class="text-sm font-medium text-foreground"
            >
              Simple House
            </span>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-3 gap-8"
    >
      <div
        class="lg:col-span-2 space-y-6"
      >
        <div
          class="relative h-[400px] md:h-[500px] rounded-lg overflow-hidden bg-secondary/10"
        >
          <img
            alt="Simple House"
            class="object-cover"
            sizes="(max-width: 768px) 100vw, 66vw"
            src="https://example.com/simple.jpg"
          />
        </div>
        <div>
          <h1
            class="text-3xl md:text-4xl font-bold text-foreground mb-4"
          >
            Simple House
          </h1>
          <div
            class="mb-6"
          >
            <span
              class="font-semibold text-accent font-cairo mb-3"
            >
              $250,000
            </span>
            <div
              class="flex items-start gap-2 undefined"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-map-pin w-5 h-5 text-secondary flex-shrink-0 mt-0.5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
              <div
                class="flex-1 min-w-0"
              >
                <p
                  class="text-sm text-secondary line-clamp-2"
                >
                  123 Simple St
                </p>
                <p
                  class="text-sm font-medium text-foreground mt-0.5"
                >
                  Simple City
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="card-elevated p-6"
        >
          <h2
            class="text-2xl font-semibold text-foreground mb-4"
          >
            Description
          </h2>
          <p
            class="text-secondary leading-relaxed"
          >
            This beautiful 
            property
             is located in 
            Simple City
            , offering 
            spacious living areas
            . Perfect for those seeking a comfortable and elegant living experience in a prime location.
          </p>
        </div>
        <div
          class="card-elevated p-6"
        >
          <h2
            class="text-2xl font-semibold text-foreground mb-4"
          >
            Property Details
          </h2>
          <dl
            class="grid grid-cols-1 md:grid-cols-2 gap-4"
          >
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Property ID
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                prop-minimal
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Location
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                Simple City
              </dd>
            </div>
            <div
              class="border-b border-border pb-3"
            >
              <dt
                class="text-sm text-secondary mb-1"
              >
                Address
              </dt>
              <dd
                class="font-medium text-foreground"
              >
                123 Simple St
              </dd>
            </div>
          </dl>
        </div>
      </div>
      <div
        class="lg:col-span-1"
      >
        <div
          class="card-elevated p-6 sticky top-8"
        >
          <div
            class="mb-6"
          >
            <span
              class="font-semibold text-accent font-cairo"
            >
              $250,000
            </span>
          </div>
          <div
            class="space-y-4 mb-6"
          >
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <span
                class="text-sm text-secondary"
              >
                Property Type
              </span>
              <span
                class="font-medium text-foreground"
              >
                N/A
              </span>
            </div>
            <div
              class="flex items-center justify-between py-3 border-b border-border"
            >
              <span
                class="text-sm text-secondary"
              >
                Location
              </span>
              <span
                class="font-medium text-foreground"
              >
                Simple City
              </span>
            </div>
          </div>
          <button
            aria-label="Contact about this property"
            class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed bg-accent text-white hover:bg-accent/90 focus:ring-accent h-10 px-4 text-sm w-full mb-3"
          >
            Contact Agent
          </button>
          <button
            aria-label="Schedule a viewing"
            class="inline-flex items-center justify-center rounded-md font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:opacity-50 disabled:cursor-not-allowed border border-border text-foreground hover:bg-secondary/5 focus:ring-accent h-10 px-4 text-sm w-full"
          >
            Schedule Viewing
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
